File: 1weathermachine.cpp
#include <fstream>
#include <string>
#include <regex>

int sum_of_numbers(std::string file_path) {
    std::ifstream file(file_path);
    std::string line;
    std::regex e("\\d+");
    int total_sum = 0;

    while (std::getline(file, line)) {
        auto begin = std::sregex_iterator(line.begin(), line.end(), e);
        auto end = std::sregex_iterator();

        if (begin != end) {
            std::string first_digit(1, (*begin).str()[0]);
            std::string last_digit(1, (*(--end)).str().back());
            total_sum += std::stoi(first_digit + last_digit);
        }
    }

    return total_sum;
}

File: 1weathermachine.py
import re

def sum_of_nums(file_path):
    with open(file_path, 'r') as file:
        lines = file.readlines()
    total_sum = 0
    for line in lines:
        numbers = re.findall('\d+', line)
        if numbers:
            first_digit = numbers[0][0]
            last_digit = numbers[-1][-1]
            total_sum += int(first_digit + last_digit)
    return total_sum

print(sum_of_nums('1weathermachine.txt'))


# i: many lines of text (num/char)
# o: sum of each line's first and last num joined, if only 1 num present it's 'numnum'

# input 1weathermachine.txt
# output 55172